jar-infer/jar-infer-lib/src/test/java/com/uber/nullaway/jarinfer/AnnotationComparator.java

```diff
@@ -0,0 +1,141 @@
+package com.uber.nullaway.jarinfer;
+
+import com.google.common.base.Preconditions;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Enumeration;
+import java.util.List;
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
+import jdk.internal.org.objectweb.asm.ClassReader;
+import jdk.internal.org.objectweb.asm.tree.AnnotationNode;
+import jdk.internal.org.objectweb.asm.tree.ClassNode;
+import jdk.internal.org.objectweb.asm.tree.MethodNode;
+
+// This class is used to compare the method annotations on the given class / jar files.
+public class AnnotationComparator {
+  public static boolean CompareMethodAnnotationsInJars(String jarFile1, String jarFile2)
+      throws IOException {
+    Preconditions.checkArgument(
+        jarFile1.endsWith(".jar") && jarFile2.endsWith(".jar"),
+        "invalid jar files: " + jarFile1 + " " + jarFile2);
+    JarFile jar1 = new JarFile(jarFile1);
+    JarFile jar2 = new JarFile(jarFile2);
+    Enumeration<JarEntry> entries1 = jar1.entries();
+    Enumeration<JarEntry> entries2 = jar2.entries();
+    while (entries1.hasMoreElements()) {
+      if (!entries2.hasMoreElements()) {
+        return false;
+      }
+      JarEntry entry1 = entries1.nextElement();
+      JarEntry entry2 = entries2.nextElement();
+      if (!entry1.getName().equals(entry2.getName())) {
+        System.out.println(
+            "  Jar entries don't match: " + entry1.getName() + " " + entry2.getName());
+        return false;
+      }
+      if (entry1.getName().endsWith(".class")) {
+        if (!CompareMethodAnnotationsInClasses(
+            jar1.getInputStream(entry1), jar2.getInputStream(entry2))) {
+          System.out.println(
+              "  Method annotations don't match in " + entry1.getName() + " " + entry2.getName());
+          return false;
+        }
+      }
+    }
+    return !entries2.hasMoreElements();
+  }
+
+  public static boolean CompareMethodAnnotationsInClasses(String classFile1, String classFile2)
+      throws IOException {
+    Preconditions.checkArgument(
+        classFile1.endsWith(".class") && classFile2.endsWith(".class"),
+        "invalid class files: " + classFile1 + " " + classFile2);
+    System.out.println("File1: " + classFile1);
+    System.out.println("File2: " + classFile2);
```

Comment:
So, I know JarInfer is already pretty verbose with print debugging, so I am fine with keeping these for now, but I am not a fan of being that verbose with unconditional prints on landed code (even the `DEBUG` flag conditional prints are probably not the best way of handling debugging for this, but not 100% sure what to do about it)

Reply:
This class is removed now.